app/models/user.rb

```diff
@@ -9,4 +9,30 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  has_many :friendships
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
+
+  def friends
+    friends_list = friendships.map { |friendship| friendship.friend if friendship.status }
+    friends_list.compact
+  end
+
+  def pending_friends
+    friendships.map { |friendship| friendship.friend unless friendship.status }.compact
+  end
+
+  def friendship_requests
+    inverse_friendships.map { |friendship| friendship.user unless friendship.status }.compact
+  end
```

Comment:
Similar to the `pending_friendships` association, to turn this method into an association you will need:
- `has_many :inverse_friends`. Where you select only the friendships that have the `status: false`. HINT: you need to declare the `class` and `foreign_key` to set this association.
-  `has_many :friendship_requests`. HINT: this is a through association that uses the previous one.